src/com/google/enterprise/gsafeed/Acl.java

```diff
@@ -47,6 +48,41 @@
   protected String inheritFrom;
   protected List<Principal> principal;
 
+  /** Inheritance types. */
+  public static enum InheritanceType {
+    CHILD_OVERRIDES("child-overrides"),
+    PARENT_OVERRIDES("parent-overrides"),
+    AND_BOTH_PERMIT("and-both-permit"),
+    LEAF_NODE("leaf-node");
+
+    private String xmlValue;
+
+    private InheritanceType(String xmlValue) {
+      this.xmlValue = xmlValue;
+    }
+
+    @Override
+    public String toString() {
+      return xmlValue;
+    }
+
+    public static InheritanceType fromString(String value) {
+      if (value.equals("child-overrides")) {
```

Comment:
Maybe:

    for (InheritanceType type : InheritanceType.values) {
      if (type.xmlValue.equals(value)) {
        return type;
      }
    }
    throw new IllegalArgumentException(value);